File: 7-conclusionLab/1-unitTests/junit-basic/src/test/java/CalculadoraTest.java
import org.example.Calculadora; // Importa a classe Calculadora que será testada.
import org.junit.jupiter.api.Test; // Importa a anotação @Test para definir métodos de teste.

import static org.junit.jupiter.api.Assertions.assertEquals; // Importa o método assertEquals para verificar se o resultado dos testes é o esperado.

public class CalculadoraTest { // Declara a classe de teste CalculadoraTest.

    @Test
    public void testSomar() { // Define um método de teste para o método somar da classe Calculadora.
        Calculadora calculadora = new Calculadora(); // Cria uma instância da Calculadora.
        int resultado = calculadora.somar(2, 3); // Chama o método somar com os valores 2 e 3, armazenando o resultado.
        assertEquals(5, resultado, "A soma entre 2 e 3 deve ser 5"); // Verifica se o resultado é igual a 5, com uma mensagem em caso de falha.
    }

    @Test
    public void testSomarNegativos() { // Define um método de teste para somas envolvendo números negativos.
        Calculadora calculadora = new Calculadora(); // Cria uma nova instância da Calculadora.
        int resultado = calculadora.somar(-1, -1); // Chama o método somar com os valores -1 e -1, armazenando o resultado.
        assertEquals(-2, resultado, "A soma entre -1 e -1 deve ser -2"); // Verifica se o resultado é igual a -2, com uma mensagem em caso de falha.
    }
}

File: 7-conclusionLab/1-unitTests/javascript/basic/calculadora.test.js
// Importando a função soma que vamos testar
const soma = require('./calculadora');

describe('Função soma', () => {
  // Teste para verificar se a soma de 2 e 3 resulta em 5
  test('deve somar 2 e 3 e retornar 5', () => {
    // Chamamos a função e verificamos se o resultado é o esperado
    expect(soma(2, 3)).toBe(5);
  });

  // Teste para verificar a soma de números negativos
  test('deve somar -1 e -1 e retornar -2', () => {
    expect(soma(-1, -1)).toBe(-2);
  });

  // Teste para somar um número positivo e um número negativo
  test('deve somar 5 e -3 e retornar 2', () => {
    expect(soma(5, -3)).toBe(2);
  });
});
